src/Database/LibSQLConnection.php

```diff
@@ -30,33 +31,6 @@ public function sync(): void
         $this->db->sync();
     }
 
-    public function getDb(): LibSQL
-    {
-        return $this->db->getDb();
-    }
-
-    public function getConnectionMode(): string
-    {
-        return $this->db->getConnectionMode();
-    }
-
-    public function statement($query, $bindings = []): bool
```

Comment:
Why are you removing actual implementation?

```php
    /**
     * Execute an SQL statement and return the boolean result.
     *
     * @param  string  $query
     * @param  array  $bindings
     * @return bool
     */
    public function statement($query, $bindings = [])
    {
        return $this->run($query, $bindings, function ($query, $bindings) {
            if ($this->pretending()) {
                return true;
            }

            $statement = $this->getPdo()->prepare($query);

            $this->bindValues($statement, $this->prepareBindings($bindings));

            $this->recordsHaveBeenModified();

            return $statement->execute();
        });
    }
```
This will totally mess up things further, especially if you will work with models inserts/updates.

Reply:
I had to lie with this to the `Connection` class:
```php
public function getDb(): LibSQL
{
    return $this->db->getDb();
}

public function getConnectionMode(): string
{
    return $this->db->getConnectionMode();
}

public function statement($query, $bindings = []): bool
{
    $res = $this->select($query, $bindings);

    return !empty($res);
}

public function getPdo(): LibSQLDatabase
{
    return $this->db;
}

public function getReadPdo(): LibSQLDatabase
{
    return $this->db;
}

public function select($query, $bindings = [], $useReadPdo = true)
{
    $result = (array) parent::select($query, $bindings, $useReadPdo);

    $resultArray = array_map(function ($item) {
        return (array) $item;
    }, $result);

    return $resultArray;
}
```
Because when I used this method:
```php
    /**
     * Execute an SQL statement and return the boolean result.
     *
     * @param  string  $query
     * @param  array  $bindings
     * @return bool
     */
    public function statement($query, $bindings = [])
    {
        return $this->run($query, $bindings, function ($query, $bindings) {
            if ($this->pretending()) {
                return true;
            }

            $statement = $this->getPdo()->prepare($query);

            $this->bindValues($statement, $this->prepareBindings($bindings));

            $this->recordsHaveBeenModified();

            return $statement->execute();
        });
    }
```
Database connections will use the PDO engine to perform binding, whereas libSQL itself does not have natural binding parameters like PDO.

Reply:
```
Database connections will use the PDO engine to perform binding, whereas libSQL itself does not have natural binding parameters like PDO.
```
Yes, that's right, and you are rewriting it inside your Statement class, method execute.

On my dev branch I've changed structure a little, to make concerns for statement/connection classes more clear.
https://github.com/tursodatabase/turso-driver-laravel/compare/main...ineersa:turso-driver-laravel:dev-branch

Primary I done that to address issue with `lastInsertId` which was not working and was implemented in the wrong place, and it's actually crucial for Eloquent models.

Also I have some strange problems with bindings on INSERT currently, where TEXT column data is truncated.

But if you will try and look into my dev branch, you can see that everything is working via base methods.

Reply:
Yea problem is more fundamental, and lies in internal not compatibility between this implementation and PDO classes.
https://github.com/tursodatabase/turso-driver-laravel/issues/14 this one has roots from there also, I will try to look at it on the next week

Reply:
So, now it's clearer why I made a lot of unconventional things with this package.

And for reason about the **connection string** need to follow the Turso conventional, because the binary itself is have a logic to identifying the connection mode like this:
```rust
pub fn get_mode(
    url: Option<String>,
    auth_token: Option<String>,
    sync_url: Option<String>,
) -> String {
    match (url, auth_token, sync_url) {
        (Some(ref url), Some(ref auth_token), Some(ref sync_url))
            if (url.starts_with("file:") || url.ends_with(".db") || url.starts_with("libsql:"))
                && !auth_token.is_empty()
                && (sync_url.starts_with("libsql://")
                    || sync_url.starts_with("http://")
                    || sync_url.starts_with("https://")) =>
        {
            "remote_replica".to_string()
        }
        (Some(ref url), Some(ref auth_token), _)
            if !auth_token.is_empty() && url.starts_with("libsql://")
                || url.starts_with("http://")
                || url.starts_with("https://") =>
        {
            "remote".to_string()
        }
        (Some(ref url), _, _)
            if url.starts_with("file:")
                || url.ends_with(".db")
                || url.starts_with("libsql:")
                || url.contains(":memory:") =>
        {
            "local".to_string()
        }
        _ => "Mode is not available!".to_string(),
    }
}
```
Check if is **Remote Replica Connection**?
- `url` starts with `file:`, ends with `.db`, or starts with `libsql:`.
- `auth_token` is not empty.
- `sync_url` starts with `libsql://`, `http://`, or `https://`.

Check if is **Remote Connection**?
- `auth_token` is not empty.
- `url` starts with `libsql://`, `http://`, or `https://`.

Check if is **Local Connection**?
- `url` starts with `file:`, ends with `.db`, starts with `libsql:`, or contains `:memory:`.

Reply:
> Yea problem is more fundamental, and lies in internal not compatibility between this implementation and PDO classes.

Please remember, the real challenge is to make it compatible without using PDO. So far what I've done is freestyle which may look very strange.